### ISSUE TITLE
[Feature] Manage VPN connections

### ISSUE THREAD

**Issue by Cu3PO42** [end-4/dots-hyprland#513]
**What would you like to be added?**

I'd love for the control panel on the right to also offer the option to connect to VPNs cofigured via Network Manager. I imagine it looking very similar to the menu already present for Wi-Fi connections, where clicking one would trigger a connection.

**How will it help**

I have several (corporate) VPNs that I might connect to (and disconnect from) multiple times a day. I currently use nm-applet for VPNs, but since Illogical Impulse already covers other networking, it would be nice to centraiize all network config there.

**Comment by Ghanshyam-shaktawat** [comment]
hey greate idea. I also needed something like this as i have stuff i do with vpn. So I create my own version of it. I just create a toggle button on the right side menu next to the toggle wifi button which use the ags networking widget to connect to a praticular vpn connection. The vpn uuid is needed to be used for connection which i stored in userOptions.ags file. If u want i can share the code as a temperarly solution.

Altough about listening the vpn connections on the sidebar like wifi  is not hard to do.  Using nmcli we get get different vpn's and there uuid's which we can used to connect to with the help of ags network widget. I can implementing it and making a pr but i got exams and will be free after 3 June.

**Comment by Cu3PO42** [comment]
You don't even need to invoke `nmcli`, you can directly use the NM library via GIR. It's also wrapped by AGS. What will take some care is that connecting to VPNs can also incur various prompts, for example providing credentials for external PKCS11 devices.

**Comment by pandalec** [comment]
Yeah like GNOME is doing it, e.g. https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/6656

**Comment by hoovad** [comment]
https://github.com/end-4/dots-hyprland/pull/1813
?

**Comment by mambucodev** [comment]
+1
